test/integration/http_integration.cc

```diff
@@ -1063,6 +1064,136 @@ void HttpIntegrationTest::testHittingEncoderFilterLimit() {
   EXPECT_STREQ("500", response->headers().Status()->value().c_str());
 }
 
+// Verifies metadata can be sent at different locations of the responses.
+void HttpIntegrationTest::testEnvoyProxyMetadataInResponse() {
```

Comment:
Can we have one test which has multiple metadata frames, and make sure all make it through?
Also one where we send invalid metadata (too large or otherwise corrupted) and make sure it's handled?

Reply:
Good point! Done.